src/node_sqlite.cc

```diff
@@ -94,9 +100,19 @@ DatabaseSync::DatabaseSync(Environment* env,
   }
 }
 
+void DatabaseSync::DeleteSessions() {
+  // all attached sessions need to be deleted before the database is closed
+  // https://www.sqlite.org/session/sqlite3session_create.html
+  for (auto* session : sessions_) {
+    sqlite3session_delete(session);
+  }
+  sessions_.clear();
```

Comment:
If the `sqlite3_session*` can be wrapped into an `std::unique_ptr` with a custom deleter then we don't need to iterate over them and delete before clearing. Would be a bit safer.

Reply:
I think that would go against the expectations of a `std::unique_ptr`, since a `Session` object also have that same `sqlite3_session*`.

Reply:
Ah, I missed that they are shared. Can these use `std::shared_ptr` in some way?

Reply:
A `std::shared_ptr` calls a deleter when all owners have let go. But in this case we need to make sure that whenever any owner wants to let go (when the session or database is closed), the session is deleted. Maybe this can be refactored with a weak_ptr at some point, but the Statement class in this file uses a similar pattern.